docs/src/rules/no-inner-declarations.md

```diff
@@ -195,6 +212,128 @@ class C {
 
 :::
 
+### blockScopedFunctions
+
+Example of **incorrect** code for this rule with `{ blockScopedFunctions: "disallow" }` option with `ecmaVersion: 2015`:
+
+::: incorrect { "sourceType": "script", "ecmaVersion": 2015 }
+
+```js
+/*eslint no-inner-declarations: ["error", { blockScopedFunctions: "disallow" }]*/
```

Comment:
```suggestion
/*eslint no-inner-declarations: ["error", "functions", { blockScopedFunctions: "disallow" }]*/
```

Without the string option, this would be an invalid configuration for this rule.